galleries/examples/images_contours_and_fields/image_antialiasing.py

```diff
@@ -45,71 +40,213 @@
+# colormap, it is what we perceive when a blue and red stripe are close to each
+# other.
+#
+# The default for the *interpolation* keyword argument is 'antialiased' which
+# will choose a Hanning filter if the image is being down-sampled or up-sampled
+# by less than a factor of three.  The default *interpolation_stage* keyword
+# argument is also 'antialiased', and for images that are down-sampled or
+# up-sampled by less than a factor of three it defaults to 'rgba'
+# interpolation.
+#
+# Anti-aliasing filtering is needed, even when up-sampling. The following image
+# up-samples 450 data pixels to 530 rendered pixels. You may note a grid of
+# line-like artifacts which stem from the extra pixels that had to be made up.
+# Since interpolation is 'nearest' they are the same as a neighboring line of
+# pixels and thus stretch the image locally so that it looks distorted.
+
 fig, ax = plt.subplots(figsize=(6.8, 6.8))
-ax.imshow(a, interpolation='nearest', cmap='gray')
-ax.set_title("upsampled by factor a 1.048, interpolation='nearest'")
-plt.show()
+ax.imshow(alarge, interpolation='nearest', cmap='grey')
+ax.set_title("upsampled by factor a 1.17, interpolation='nearest'")
 
 # %%
 # Better antialiasing algorithms can reduce this effect:
 fig, ax = plt.subplots(figsize=(6.8, 6.8))
-ax.imshow(a, interpolation='antialiased', cmap='gray')
-ax.set_title("upsampled by factor a 1.048, interpolation='antialiased'")
-plt.show()
+ax.imshow(alarge, interpolation='antialiased', cmap='grey')
+ax.set_title("upsampled by factor a 1.17, interpolation='antialiased'")
 
 # %%
 # Apart from the default 'hanning' antialiasing, `~.Axes.imshow` supports a
 # number of different interpolation algorithms, which may work better or
-# worse depending on the pattern.
+# worse depending on the underlying data.
 fig, axs = plt.subplots(1, 2, figsize=(7, 4), layout='constrained')
 for ax, interp in zip(axs, ['hanning', 'lanczos']):
-    ax.imshow(a, interpolation=interp, cmap='gray')
+    ax.imshow(alarge, interpolation=interp, cmap='gray')
     ax.set_title(f"interpolation='{interp}'")
+
+# %%
+# A final example shows the desirability of performing the anti-aliasing at
+# the RGBA stage.  In the following, the data in the upper 100 rows is exactly
+# 0.0, and data in the inner circle is exactly 2.0. If we perform the
+# *interpolation_stage* in 'data' space and use an anti-aliasing filter (first
+# panel), then floating point imprecision makes some of the data values just a
+# bit less than zero or a bit more than 2.0, and they get assigned the under-
+# or over- colors. This can be avoided if you do not use an anti-aliasing filter
+# (*interpolation* set set to 'nearest'), however, that makes the part of the
+# data susceptible to Moiré patterns much worse (second panel).  Therefore, we
+# recommend the default *interpolation* of 'hanning'/'antialiased', and
+# *interpolation_stage* of 'rgba'/'antialiased' for most down-sampling
+# situations (last panel).
```

Comment:
```suggestion
# recommend the either using default *interpolation* of 'hanning'/'antialiased', 
# and *interpolation_stage* of 'rgba'/'antialiased' for most down-sampling
# situations (last panel) or *interpolation* of 'nearest' and *interpolation_stage*
# of 'data'.
```

Reply:
This seems more confusing than helpful.  We've already said what the different arguments do, and we have a default that we recommend.